.github/workflows/inspektor-gadget.yml

```diff
@@ -1344,6 +1347,20 @@ jobs:
           GADGET_TAG=${{ steps.set-repo-determine-image-tag.outputs.gadget-tag }} \
           BUILDER_IMAGE=${{ needs.build-helper-images.outputs.ebpf_builder_image }} \
           push-gadgets -o install/ig
+      - name: Install Cosign
+        if: needs.check-secrets.outputs.cosign == 'true' && ${{ github.event_name }} == 'push'
+        uses: sigstore/cosign-installer@main
+      - name: Sign the manifest list
+        if: needs.check-secrets.outputs.cosign == 'true' && ${{ github.event_name }} == 'push'
+        env:
+          COSIGN_PASSWORD: '${{ secrets.COSIGN_PASSWORD }}'
+          COSIGN_PRIVATE_KEY: '${{ secrets.COSIGN_PRIVATE_KEY }}'
+        run: |
+          for i in $(make -C gadgets --no-print-directory list_gadgets); do
+            gadget="${{ steps.set-repo-determine-image-tag.outputs.gadget-repository }}/${i}:${{ steps.set-repo-determine-image-tag.outputs.gadget-tag }}"
+
+            cosign sign --key env://COSIGN_PRIVATE_KEY --yes --recursive $gadget
```

Comment:
The following warning message is displayed:

```
WARNING: Image reference ttl.sh/fb9571407fc55a27104917d3d3e7a36ee8735e0e/trace_dns:2678-merge uses a tag, not a digest, to identify the image to sign.
    This can lead you to sign a different image than the intended one. Please use a
    digest (example.com/ubuntu@sha256:abc123...) rather than tag
    (example.com/ubuntu:latest) for the input to cosign. The ability to refer to
    images by tag will be removed in a future release.
```

We can manage this in another PR, but please create an issue for it.